test/end-to-end/core.sh

```diff
@@ -53,7 +53,16 @@ os::log::info "oc version:        `oc version`"
 
 # service dns entry is visible via master service
 # find the IP of the master service by asking the API_HOST to verify DNS is running there
+# might need to wait a bit to ensure the dns cache is primed
+os::cmd::try_until_text "dig "@${API_HOST}" "kubernetes.default.svc.cluster.local." +short A | head -n 1" "^.+$"
+# TODO REMOVE THIS HACK
+while true; do
```

Comment:
@stevekuznetsov for the rescue! We can't loop here forever, at least a reasonable timeout with a failure would be needed.